addons/web_dashboard/models/res_users.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime, timedelta
+
+from openerp import models, api, _
+from openerp.exceptions import RedirectWarning
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class Users(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def web_dashboard_create_users(self, emails, optional_message=""):
+        existing_users = self.search(['&', ('login', 'in', emails), '|', ('active', '=', True), ('active', '=', False)])
+
+        # Processing existing users
+        for user in existing_users:
+            if not user.login_date and not user.signup_valid:
+                user.action_reset_password()
+            else:
+                # Already a user, reactivated in case it is deactivated
+                user.active = True
+        new_emails = set(emails) - set(existing_users.mapped('login'))
```

Comment:
we always displayed login as a email address. And user created through dashboard will have email in login.
![download](https://cloud.githubusercontent.com/assets/7629089/9577084/3c323822-4ffb-11e5-8acb-a8f49d3e90d8.png)
But it is not in every case(e.g. user's demo data has admin, demo etc :grin: ). so we can change to email, but there will be chances of someone make change in email id because email is related field(partner_id.email)